### PR TITLE
Update node release based on maplibre-js release workflow - v2 with changelog

### PR DESCRIPTION
**Pull request by acalcutt**
This PR updates the node release workflow so it no longer increments the version on it's own. This should allow it to work with the approval requirements on this repo. I've made this work more like the maplibre-js workflow at https://github.com/maplibre/maplibre-gl-js/blob/b7b86c1018842a139d4ced21a071ca4a3d9f24b5/.github/workflows/release.yml .

To increment the version now, someone would make a PR that changes the version in 'package.json' and release notes in 'platform\node\CHANGELOG.md'. Once the version in package.json has change, the workflow automatically checks that version against what is published on npm. If the version has not been published, the binaries get built and published to github using NODE_PRE_GYP_GUTHUB. Then the release is published to npm. 

I've also cherry picked some https://github.com/maplibre/maplibre-native/pull/1941 I made in the opengl-2 branch.

- update '@acalcutt/node-pre-gyp' to support node 20
- put '--fallback-to-build=false' in package.json. I had changed this a while back but it should be false since we don't have GYP build files
- fix 'runner.arch' that should be 'matrix.arch'
- update readme to reflect binaries being built
- remove unused node-ci-mac.yml
- I was also thinking about versioning. Maybe in the 'opengl-2' branch we can continue a 5.x release , but in main we can start with '6.0.0-pre.0' (once we get something together for a metal pre-release)

**Other fixes**
- Added automatic inserting of node changelog into release notes.
- Try to use GITHUB_TOKEN instead of NODE_PRE_GYP_GITHUB_TOKEN. Added 'contents: write' permission to allow this token to publish the release and add release notes.

**Testing**
I have created a test release (minus macos support) at
https://github.com/WifiDB/maplibre-native/actions/runs/7429441972
https://www.npmjs.com/package/@acalcutt/maplibre-gl-native-test/v/5.2.9
https://github.com/WifiDB/maplibre-native/releases/tag/node-v5.2.9

**Notes**
After this is approved I plan to cherry pick these changes over to 'opengl-2' so I could also make a updated release with macos support there, so a squash and merge may be easier if this is merged.

### CODE REVIEW

**Review by louwers — APPROVED** [review]
OK looks good, let's try it!

Maybe you want to cancel a release if no changelog was found, but up to you.

**Comment by acalcutt** [comment]
I am wondering what you think about how to test this after it is merged. After this is merged, should I cherry-pick it to 'opengl-2' and test it there? or should I test it in main, minus macos support (maybe with a 6.0.0-pre.0 version and notes it does not work on macos)

Really right now I want the release from 'opengl-2' with node 20 support + macos still working.

**Comment by acalcutt** [comment]
I was thinking there was another reason I had moved the update of release notes to after the binaries got published that I had forgotten.

When 'node-pre-gyp-github' publishes the binaries, it was overwriting the text of the release with the npm package name ( https://github.com/acalcutt/node-pre-gyp-github/blob/master/index.js#L73 ). So if I did it before the changelog text would get wiped out.

I suppose I could just check if it is empty in the first job, when it checks if the npm version is published.

**Comment by louwers** [comment]
> Really right now I want the release from 'opengl-2' with node 20 support + macos still working.

Sure! Makes sense then.

File: platform/node/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Instructions for making a Node release
```
Comment (louwers) [inline platform/node/RELEASE.md:1]: Great!

**Pull request merged**